internal/xds/translator/ratelimit.go

```diff
@@ -715,8 +817,17 @@ func (t *Translator) createRateLimitServiceCluster(tCtx *types.ResourceVersionTa
 	})
 }
 
-func getDomainName(route *ir.HTTPRoute) string {
-	return strings.Replace(route.Traffic.Name, "/", "-", 1)
+// getDomainSharedName returns the shared domain (stripped policy name) if any shared rule exists,
+// otherwise returns the listener name.
+func getDomainSharedName(route *ir.HTTPRoute) string {
+	if isValidGlobalRateLimit(route) && len(route.Traffic.RateLimit.Global.Rules) > 0 {
+		for _, rule := range route.Traffic.RateLimit.Global.Rules {
+			if isRuleShared(rule) {
+				return stripRuleIndexSuffix(rule.Name)
```

Comment:
If we allow a user-specified name for rules, this won't work.

Instead, we can add a name field to ir.GlobalRateLimit, and set its value to "namespace/policy-name".